vcd/resource_vcd_ip_space.go

```diff
@@ -0,0 +1,429 @@
+		orgName := resourceURI[0]
+
+		org, err := vcdClient.GetOrgByName(orgName)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving Org '%s': %s", orgName, err)
+		}
+
+		ipSpace, err = vcdClient.GetIpSpaceByNameAndOrgId(ipSpaceName, org.Org.ID)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving IP Space '%s' in Org '%s': %s", ipSpaceName, orgName, err)
+		}
+		dSet(d, "org_id", org.Org.ID)
+	case len(resourceURI) == 1: // Resource path is supplied as `ip-space-name`
+		var err error
+		ipSpaceName := resourceURI[0]
+		ipSpace, err = vcdClient.GetIpSpaceByName(ipSpaceName)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving IP Space '%s': %s", ipSpaceName, err)
+		}
+	default:
+		return nil, fmt.Errorf("unrecognized path for IP Space import '%s'", d.Id())
+	}
+
+	d.SetId(ipSpace.IpSpace.ID)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func getIpSpaceType(d *schema.ResourceData) (*types.IpSpace, error) {
+	ipSpace := &types.IpSpace{
+		Name:                      d.Get("name").(string),
+		Description:               d.Get("description").(string),
+		Type:                      d.Get("type").(string),
+		IPSpaceInternalScope:      convertSchemaSetToSliceOfStrings(d.Get("internal_scope").(*schema.Set)),
+		IPSpaceExternalScope:      d.Get("external_scope").(string),
+		RouteAdvertisementEnabled: d.Get("route_advertisement_enabled").(bool),
+	}
+
+	// IP Ranges
+	ipRangeQuota := d.Get("ip_range_quota").(string)
+	ipRangeQuotaInt, _ := strconv.Atoi(ipRangeQuota) // error is ignored because validation is enforced in schema
+
+	ipSpace.IPSpaceRanges = types.IPSpaceRanges{
+		DefaultFloatingIPQuota: ipRangeQuotaInt,
+	}
+
+	ipRanges := d.Get("ip_range").(*schema.Set)
+	ipRangesSlice := ipRanges.List()
+	if len(ipRangesSlice) > 0 {
+		ipSpace.IPSpaceRanges.IPRanges = make([]types.IpSpaceRangeValues, len(ipRangesSlice))
+		for ipRangeIndex := range ipRangesSlice {
+			ipRangeStrings := convertToStringMap(ipRangesSlice[ipRangeIndex].(map[string]interface{}))
+
+			ipSpace.IPSpaceRanges.IPRanges[ipRangeIndex].StartIPAddress = ipRangeStrings["start_address"]
+			ipSpace.IPSpaceRanges.IPRanges[ipRangeIndex].EndIPAddress = ipRangeStrings["end_address"]
+		}
+	}
+	// EOF IP Ranges
+
+	// IP Prefixes
+	ipPrefixes := d.Get("ip_prefix").(*schema.Set)
+	ipPrefixesSlice := ipPrefixes.List()
+
+	if len(ipPrefixesSlice) > 0 {
+		ipSpace.IPSpacePrefixes = []types.IPSpacePrefixes{}
+	}
+
+	for ipPrefixIndex := range ipPrefixesSlice {
+		singleIpPrefix := ipPrefixesSlice[ipPrefixIndex]
+		ipPrefixMap := singleIpPrefix.(map[string]interface{})
+		ipPrefixQuota := ipPrefixMap["default_quota"].(string)
+		ipPrefixQuotaInt, _ := strconv.Atoi(ipPrefixQuota) // ignoring error as validation is enforce in schema
+
+		ipSpacePrefixType := types.IPSpacePrefixes{
+			DefaultQuotaForPrefixLength: ipPrefixQuotaInt,
+		}
+
+		// Extract IP prefixes
+		// 'prefix'
+		ipPrefixPrefix := ipPrefixMap["prefix"].(*schema.Set)
```

Comment:
Comment is unfinished?
Also, why the double `...PrefixPrefix...` in the variable names? Could there be an alternative naming? Looks peculiar :)

Reply:
It is the nesting of the field in the API structure that caused me to use such a name - the address of the field happens to be `ip_prefix.X.prefix......`

I have tried to update the comment as well as change the name of that variable to begin with `nestedPrefix***`